bin/inx-chronicle/src/api/stardust/indexer/routes.rs

```diff
@@ -106,3 +123,101 @@ async fn alias_outputs(
         cursor,
     })
 }
+
+async fn output_by_foundry_id(
+    database: Extension<MongoDb>,
+    Path(foundry_id): Path<String>,
+) -> ApiResult<IndexerOutputResponse> {
+    let foundry_id = FoundryId::from_str(&foundry_id)?;
+    let res = database
+        .get_foundry_output_by_id(foundry_id)
+        .await?
+        .ok_or(ApiError::NoResults)?;
+    Ok(IndexerOutputResponse {
+        ledger_index: res.ledger_index.0,
+        output_id: res.output_id.to_hex(),
+    })
+}
+
+async fn foundry_outputs(
+    database: Extension<MongoDb>,
+    FoundryOutputsPagination {
+        query,
+        page_size,
+        cursor,
+    }: FoundryOutputsPagination,
+) -> ApiResult<IndexerOutputsResponse> {
+    let res = database
+        .get_foundry_outputs(
+            query,
+            // Get one extra record so that we can create the cursor.
+            page_size + 1,
+            cursor.map(|(ms, o)| (ms.into(), o)),
+        )
+        .await?
+        .ok_or(ApiError::NoResults)?;
+    let mut iter = res.outputs.iter();
+
+    // Take all of the requested records first
+    let items = iter.by_ref().take(page_size).map(|o| o.output_id.to_hex()).collect();
+
+    // If any record is left, use it to make the cursor
+    let cursor = iter
+        .next()
+        .map(|rec| format!("{}.{}.{}", rec.booked_index, rec.output_id.to_hex(), page_size));
```

Comment:
Would we save some code if this where a helper function that is generic over the `Output` within the stream?